### PR TITLE
docs: update wallet management docs

### PR DESCRIPTION
**Pull request by antbern**
## Motivation and Context

- Update the Wallet management chapter to include information about SSS, remove stuff that is no longer valid and add links to API reference. Replaced code snippets with links to examples 💯 
- fix some links
- Fix Licensing chapter.
- Strip some trailing whitespace.

## Checklist

Please ensure that your PR meets the following requirements:

### General

- [ ] ~~Code follows project style and guidelines.~~
- [ ] ~~No redundant or duplicate code.~~
- [ ] ~~No added new dependencies without clear justification and approval.~~

### Documentation

- [x] Added/updated relevant documentation (e.g., README, inline comments).
- [ ] Added a CHANGELOG entry for major changes.

### Testing

- [ ] ~~Added relevant test cases, leveraging tools like `rstest` or similar for reusable tests.~~
- [ ] ~~Tests pass locally.~~

### Build & CI/CD

- [x] Confirmed no CI/CD pipeline issues.
- [x] Updated build scripts where necessary.

## Comments to Reviewer

Please provide any comments or points of consideration for the reviewers, such as specific areas of focus, potential edge cases, or context that would aid the review process.

### Reviewer Responsibilities

- [ ] Code adheres to style and guidelines.
- [ ] All tests are appropriately covered and they pass.
- [ ] Consider performance and/or security impacts.
- [ ] Constructive feedback is provided.

### CODE REVIEW

**Review by kristifidani — APPROVED** [review]
Looks so much better with all the updated information and references to our examples and API reference. Nice 💯

**Review by hauju — APPROVED** [review]

File: docs/docs/SDK Reference/SDK API Reference.md [hunk 1]
```diff
@@ -109,7 +109,7 @@ The Rustdoc API reference is available [here](../rust-docs/doc/etopay_sdk/index.
 
 | Method | Arguments | Returns | Dependencies | Level | Repeat|
 |--------|-----------|---------|--------------|-------|-------|
-| Sets the network | `network_id` - The network_id as string.| | [Constructor](./SDK%20API%20Reference.md#instantiating-the-sdk) | Basic | Handle |
+| Sets the network | `network_key` - The key of the network as s string.| | [Constructor](./SDK%20API%20Reference.md#instantiating-the-sdk) | Basic | Handle |
```
Comment (mighty840) [inline docs/docs/SDK Reference/SDK API Reference.md:112]: Is there one s too many in string?

Comment (antbern) [reply docs/docs/SDK Reference/SDK API Reference.md:112]: Good catch! It should actually be `a` :
```suggestion
| Sets the network | `network_key` - The key of the network as a string.| | [Constructor](./SDK%20API%20Reference.md#instantiating-the-sdk) | Basic | Handle |
```

**Review by mighty840 — APPROVED** [review]
